### ISSUE TITLE
run

### ISSUE THREAD

**Issue by Rahul-kateith** [yueyu1030/STEAM#1]
I have trained the model.
How will I generate taxonomy now for some particular keywords.

**Comment by Rahul-kateith** [comment]
I ran bert_emb_extractor.py and got vector embeddings of shape 768, but can we use it to create taxonomy terms.

**Comment by yueyu1030** [comment]
Hi,
I am not sure about the meaning of `create taxonomy terms' here. In our experiments, the train/dev/test terms are given and split from the original dataset, so adding completely new terms is beyond the topic in this paper.

If you want to add extra new terms, you can try to use some NER tools to extract key entities first, then you can use the method described in our paper for expansion. Tell me if you have other questions, thanks!

**Comment by yueyu1030** [comment]
You can also use Glove and word2vec embeddings, but we find that the empirical performance of Glove and word2vec is not so good as BERT.

**Comment by Rahul-kateith** [comment]
Thnx for the reply.
My main motive is to create taxonomy terms for telecommunications. What changes should I make?

**Comment by yueyu1030** [comment]
If you want to expand the existing telecommunication communities, we recommend that you need to first obtain a set of new terms, and then you can try to use our method to attach these terms into the existing taxonomy.

**Comment by Rahul-kateith** [comment]
But the end result is in vector form..... How will I get those terms that
can be used for taxonomy

**Comment by yueyu1030** [comment]
The vector stands for the probability for the test term being attached to a specific term in the mini-path. You can refer to section 3.3. in the original paper  for how to determine the correct position.

**Comment by Rahul-kateith** [comment]
Also, this is the dataset, which files need to be updated?
Like I have to add my telecoms wordings to which folder?
Thanks.

**Comment by yueyu1030** [comment]
Please see the function train_test_data_load in `utils_path.py' for detail information. 
Specifically, you need to first extract mini-path from the existing taxonomy, and then prepare the embedding/dependency path/lexico-syntactic features as instructed in the main repo. After that, you can run your code with telecoms wordings.

**Comment by Rahul-kateith** [comment]
I went through the function. What paths does paths.json and
paths_index_direct.json contain?
 It seems as if the taxonomy is already generated.

**Comment by Rahul-kateith** [comment]
My task is to generate taxonomy just by random keywords or a paragraph.

**Comment by yueyu1030** [comment]
Ahh, in our problem there is a taxonomy expansion problem where a seed taxonomy is given. For your task, maybe you can refer to https://www.aclweb.org/anthology/S16-1206/ for reference. Their method might be more suitable.

**Comment by Rahul-kateith** [comment]
Thnx for the link.
Can you give any way for this? Like I have obtained all the terms that can
be used to create taxonomy but how to decide which term can be a parent
node and which can be a child node?

**Comment by yueyu1030** [comment]
In the case where a seed taxonomy is not available, we suggest there are two ways to do that:
(1) You can try to use an unsupervised taxonomy expansion/hypernymy detection approach to do that such as https://www.ijcai.org/Proceedings/15/Papers/206.pdf.
(2) You can try to first train a hypernymy detection neural network on some existing taxonomies (such as wordnet/semeval etc.) and then use the network for predicting the child/parent relationships. Note that this may suffer from domain shift.
Hope this will be helpful.

**Comment by Rahul-kateith** [comment]
Okay thnx buddy for the help.

**Comment by yueyu1030** [comment]
No problem. Now I am going to close this issue.

**Comment by Rahul-kateith** [comment]
Hey, I created a taxonomy and I am working on it's expansion now.
Can you tell the steps?

Task: Expansion of taxonomy for telecommunications.
Initial steps: With the initial taxonomy I have, I created supervised data
with labels 0(non-pairs) and 1(pairs),
What can be the next steps. Also can I use pre-trained embeddings somewhere?

**Comment by yueyu1030** [comment]
Hi, with the supervised data, the first step could be training a classifier to identify parent-child relations. Then, for each term that needs to be attached to the existing taxonomy, we can calculate the score for each term and determine the correct position.
For pre-trained embeddings, we recommend the Glove or BERT embeddings. For BERT embeddings, you can take a look at https://mccormickml.com/2019/05/14/BERT-word-embeddings-tutorial/ for procedures. For Glove Embeddings, you can take a look at https://nlp.stanford.edu/projects/glove/. Hope these will be helpful.

**Comment by Rahul-kateith** [comment]
I already built a model.
gave correct pairs a label of 1 and non-pair a label of 0.
Using cosine similarity, got similar words for both words in the pair, so I
could increase the dataset. So, for each pair, I generated nearly 50
similar pairs.
Now used spacy for phrase embeddings, thus I got a vector of length 96  and
a label attached to it. Now I am classifying this on a simple neural
network.

Is the method correct? How can it be improved?
Whether this will really learn parent-child relation?
Thanks.

**Comment by yueyu1030** [comment]
Hi, this is the simple method for learning parent-child relations.
 If you want to predict the performance, you can try to do the followings:
(1) Use word embeddings with larger dimension
(2) Consider multiple types of features. (i.e. lexico-syntactic features and relation features described in our paper)
Thanks!
